tests/test_on_data/test_metadata/test_neuralynx_metadata.py

```diff
@@ -13,56 +10,122 @@
+        assert device_metadata[-1] == {"name": "DigitalLynxSX", "description": "Cheetah 5.6.3"}
 
 
 def test_neuralynx_cheetah_v540_metadata():
+    import neo
+    import distutils.version as version
+
     folder_path = NLX_PATH / "Cheetah_v5.4.0" / "original_data"
-    assert get_metadata(str(folder_path)) == dict(
-        session_start_time=datetime.datetime(2001, 1, 1, 0, 0),
-    )
+    metadata = NeuralynxRecordingInterface(folder_path).get_metadata()
+    file_metadata = metadata["NWBFile"]
+
+    assert file_metadata["session_start_time"] == datetime.datetime(2001, 1, 1, 0, 0)
+
+    assert '"recording_closed": "2001-01-01 00:00:00"' in file_metadata["notes"]
+    assert '"ADMaxValue": "32767"' in file_metadata["notes"]
+
+    # the sampling rate and device is only exposed on neo dev as of now. TODO: Remove if after neo >=0.12
+    if version.LooseVersion(neo.__version__) > version.LooseVersion("0.11.0"):
+        assert '"sampling_rate": "1017.375"' in file_metadata["notes"]
+
+        device_metadata = metadata["Ecephys"]["Device"]
+        assert device_metadata[-1] == {"name": "DigitalLynx", "description": "Cheetah 5.4.0"}
 
 
 def test_neuralynx_filtering():
-    file_path = NLX_PATH / "Cheetah_v5.7.4" / "original_data" / "CSC1.ncs"
-    assert (
-        get_filtering(str(file_path)) == '{"DSPLowCutFilterEnabled": "True", '
-        '"DspLowCutFrequency": "10", '
-        '"DspLowCutNumTaps": "0", '
-        '"DspLowCutFilterType": "DCO", '
-        '"DSPHighCutFilterEnabled": "True", '
-        '"DspHighCutFrequency": "9000", '
-        '"DspHighCutNumTaps": "64", '
-        '"DspHighCutFilterType": "FIR", '
-        '"DspDelayCompensation": "Enabled", '
-        '"DspFilterDelay_µs": "984"}'
-    )
-
-
-def test_neuralynx_filtering_recording_extractor():
-
-    ri = NeuralynxRecordingInterface(folder_path=(NLX_PATH / "Cheetah_v5.7.4" / "original_data"))
-    filtering = ri.recording_extractor.get_property("filtering")[0][:]
-    expected_filtering = (
-        '{"DSPLowCutFilterEnabled": "True", '
-        '"DspLowCutFrequency": "10", '
-        '"DspLowCutNumTaps": "0", '
-        '"DspLowCutFilterType": "DCO", '
-        '"DSPHighCutFilterEnabled": "True", '
-        '"DspHighCutFrequency": "9000", '
-        '"DspHighCutNumTaps": "64", '
-        '"DspHighCutFilterType": "FIR", '
-        '"DspDelayCompensation": "Enabled", '
-        '"DspFilterDelay_µs": "984"}'
-    )
-    assert filtering == expected_filtering
+    file_path = NLX_PATH / "Cheetah_v5.7.4" / "original_data"
+    interface = NeuralynxRecordingInterface(file_path)
+
+    expected_single_channel_props = {
+        "DSPLowCutFilterEnabled": "True",
+        "DspLowCutFrequency": "10",
+        "DspLowCutNumTaps": "0",
+        "DspLowCutFilterType": "DCO",
+        "DSPHighCutFilterEnabled": "True",
+        "DspHighCutFrequency": "9000",
+        "DspHighCutNumTaps": "64",
+        "DspHighCutFilterType": "FIR",
+        "DspDelayCompensation": "Enabled",
```

Comment:
It looks like there are int and bool values here there encoded as strings. Is there a reason for that?

Reply:
For export of these properties we need to convert bool values to strings, as we otherwise get an error in nwb (see https://github.com/catalystneuro/neuroconv/issues/185 first checkpoint).

Reply:
the electrodes table should be able to handle ints and bools with no problem. What's the error in NWB?

Reply:
It's not an error with NWB, it's an error with channel properties that are bool which are then used in our `add_electrodes` function (see https://github.com/catalystneuro/neuroconv/issues/185) which doesn't have a default type set for that class, which is something needed when writing multiple recording objects (not all of which have that same boolean channel property)

Reply:
The `float` and `int` values here shouldn't have to be strings though